### PR TITLE
add correct default to rpcHost

### LINKED ISSUE #7721: p2p-host config leak
Pointed out by `@seamonkey82` on discord.

> Current master attempts to open --metrics-port on the --p2p-host address.

I have verified the bug was introduced in https://github.com/hyperledger/besu/commit/f4dc48d94d276f1ed504b707b8f9ccf278a5cc63

To recreate...
```
 $BESU --p2p-host=9.9.9.9
...
2024-10-04 11:34:28.916+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 9.9.9.9:8551
```

By default we start JSON-RPC service for engine API on 127.0.0.1. The bug means setting p2p-host is overwriting other host configs.

When you specify the relevant host, that correctly takes precedence...

```
$BESU --rpc-http-host=1.1.1.1 --p2p-host=9.9.9.9
2024-10-04 11:40:22.499+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 1.1.1.1:8551
```

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
set the default value for the rpc Http Host CLI option so that it gets initialized correctly

## Fixed Issue(s)
fixes #7721 

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Comment by macfarla** [comment]
With this PR

```
besu --p2p-host=9.9.9.9 --metrics-enabled

2024-10-04 14:17:10.890+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 127.0.0.1:9545
2024-10-04 14:17:10.917+10:00 | vert.x-eventloop-thread-1 | INFO  | MetricsHttpService | Metrics service started and listening on 127.0.0.1:9545
2024-10-04 14:17:10.919+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 127.0.0.1:8551
2024-10-04 14:17:10.922+10:00 | vert.x-eventloop-thread-1 | INFO  | EngineJsonRpcService | JSON-RPC service started and listening on 127.0.0.1:8551
```

**Comment by macfarla** [comment]
Without this PR
```
 besu --p2p-host=9.9.9.9 

...
2024-10-04 15:24:57.477+10:00 | main | INFO  | Runner | Starting external services ...
2024-10-04 15:24:57.478+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 9.9.9.9:8551

```

and with metrics enabled, since metrics service is initalized first

```
 besu --p2p-host=9.9.9.9 --metrics-enabled

2024-10-04 14:16:29.992+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 9.9.9.9:9545
```

**Comment by siladu** [comment]
> Without this PR
> and with metrics enabled, since metrics service is initalized first
> 
> ```
>  besu --p2p-host=9.9.9.9 --metrics-enabled
> 
> 2024-10-04 14:16:29.992+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 9.9.9.9:9545
> ```

I don't see this behaviour on the latest main

```
➜  besu-local-nodes git:(master) ✗ $BESU --p2p-host=9.9.9.9 --metrics-enabled
...
# Besu version 24.10-develop-81f9fc9                                                               #
...
2024-10-04 16:26:19.020+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 127.0.0.1:9545
2024-10-04 16:26:19.053+10:00 | vert.x-eventloop-thread-1 | INFO  | MetricsHttpService | Metrics service started and listening on 127.0.0.1:9545
2024-10-04 16:26:19.054+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 9.9.9.9:8551
```

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java [hunk 1]
```diff
@@ -1029,6 +1029,12 @@ public void p2pHostAndPortOptionsAreRespected() {
 
     assertThat(commandOutput.toString(UTF_8)).isEmpty();
     assertThat(commandErrorOutput.toString(UTF_8)).isEmpty();
+
+    // rpc host should remain default ie 127.0.0.1
+    verify(mockRunnerBuilder).jsonRpcConfiguration(jsonRpcConfigArgumentCaptor.capture());
+    verify(mockRunnerBuilder).build();
+
+    assertThat(jsonRpcConfigArgumentCaptor.getValue().getHost()).isEqualTo("127.0.0.1");
```
Comment (siladu) [inline besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:1037]: Nice 👍 

How do we know other host configs aren't affected as well?

**Review by siladu — APPROVED** [review]
It fixes the issue so will approve it.

TBH, it's not clear to me why https://github.com/hyperledger/besu/pull/7682 broke it and why this is the correct fix and so I don't have confidence there aren't other issues with https://github.com/hyperledger/besu/pull/7682

**Comment by macfarla** [comment]
> > Without this PR
> > and with metrics enabled, since metrics service is initalized first
> > ```
> >  besu --p2p-host=9.9.9.9 --metrics-enabled
> > 
> > 2024-10-04 14:16:29.992+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 9.9.9.9:9545
> > ```
> 
> I don't see this behaviour on the latest main:
> 
> ```
> $ $BESU --p2p-host=9.9.9.9 --metrics-enabled
> ...
> # Besu version 24.10-develop-81f9fc9                                                               #
> ...
> 2024-10-04 16:26:19.020+10:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 127.0.0.1:9545
> 2024-10-04 16:26:19.053+10:00 | vert.x-eventloop-thread-1 | INFO  | MetricsHttpService | Metrics service started and listening on 127.0.0.1:9545
> 2024-10-04 16:26:19.054+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 9.9.9.9:8551
> ```
> 
> Did you test before [911f12e](https://github.com/hyperledger/besu/commit/911f12efa0039dbeb76606d57d6b0329589ad5a3) perhaps?

yes have re-tested and it's only the JSON-RPC host that's affected

`
besu --p2p-host=9.9.9.9 --metrics-push-enabled
`

```
2024-10-08 11:46:30.657+10:00 | main | INFO  | MetricsPushGatewayService | Starting metrics push gateway service pushing to 127.0.0.1:9001
2024-10-08 11:46:30.659+10:00 | main | INFO  | EngineJsonRpcService | Starting JSON-RPC service on 9.9.9.9:8551
```

**Pull request merged**